doc/continuous_integration/dockerhub.md

```diff
@@ -0,0 +1,64 @@
+
+* [Navigation2 on DockerHub](https://hub.docker.com/r/rosplanning/navigation2)
+
+While DockerHub does not require the use of configuration files in the source repo, the Dockerfiles and respective scripts used to customize automated builds of images are tracked in the [`.dockerhub`](/.dockerhub) directory. These scripts include custom build phase hooks used by DockerHub during automated builds. More info on automated builds can be found here:
+
+* [How Automated Builds work](https://docs.docker.com/docker-hub/builds)
+* [Advanced options for Autobuild and Autotest](https://docs.docker.com/docker-hub/builds/advanced)
+
+Automated Builds are controlled via the build configurations menu within the repo's administrative console on DockerHub. For reference, a figure of the project's build configurations is shown here:
+
+![DockerHub Build Configurations](figs/dockerhub_build_configurations.png)
+
+Here the repo's source repository is linked to the project's GitHub repo, while Autotest is disabled given a dedicated CI service is separately used to run test jobs. Repository linking can be used so that whenever a parent image repo is updated on DockerHub, it will also trigger a build for the project's DockerHub repository. Note that this only works for non-official library images, and only for build rules where Autobuild is is enabled. Two build rules are added for the main branch, providing both a release and debug tag for CI to pull from. The relative Dockerfiles paths are designated, while the build context is intentionally left empty, ensuring the build phase hooks within the same paths are used appropriately. Build caching is also enabled to shorten image turnaround time if multiple rebuilds a day are triggered.
+
+The build hooks, e.g. [build](/.dockerhub/debug/hooks/build) , are for customizing docker build arguments in the Dockerfile via the `ARG` directives; such as changing the base `FROM_IMAGE` between release or debug tags, adjusting the colcon mixins for each workspace to enable code coverage, or disabling fail on build failure, preventing source build breakages from blocking CI image tag updates. This also allows the slower debug CI to build and test only using the default RMW, while allowing the faster release CI jobs to build and test from an image with more RMWs installed.
+
+## Autobuild
+
+While the free registry for hosting is great, allowing projects gigabytes of free storage/bandwidth to cache pre-configured images for various branches or CI scenarios, that would otherwise take CI instances far longer to rebuild from scratch rather than pull from the registry, the automated build integration for DockerHub is basic. Rather than trigging build rules via pushed commits to matching GitHub branches, Autobuild is left disabled and the Build Trigger API is used instead. This Build Trigger API is evoked from scheduled cron jobs to rate limit DockerHub rebuilds; e.g. preventing hourly merge commits from needlessly churning CI image tags.
```

Comment:
This first sentence is too long and isn't structured readably, can you please break into a couple of sentences? 

At the end of this paragraph, say how we use it: "We have it set to run once a day". You explained the "what" but not the "how we use it".